src/main/java/nl/tudelft/pixelperfect/event/parameter/EventParameterCollection.java

```diff
@@ -75,6 +76,24 @@ public String toString() {
   }
 
   /**
+   * Generate a debug string representation for the parameters.
+   * 
+   * @return the debug string.
+   */
+  public String toDebugString() {
+    StringBuilder sb = new StringBuilder();
+    if (map.isEmpty()) {
+      sb.append("None");
+    } else {
+      for (EventParameter param : map.values()) {
+        sb.append(param.getNumberValue());
```

Comment:
You should make a toDebugString method for the parameter, because sometimes a parameter does not have a numberValue. In this case you should use the value instead of the number value.